dataprofiler/profilers/profile_builder.py

```diff
@@ -642,14 +639,29 @@ def _update_profile_from_chunk(df, profile=None, sample_size=None,
         :return: list of column profile base subclasses
         :rtype: list(BaseColumnProfiler)
         """
-        
-        if not profile:
-            profile = OrderedDict()
 
         if len(df.columns) != len(df.columns.unique()):
             raise ValueError('`Profiler` does not currently support data which '
                              'contains columns with duplicate names.')
 
+        # assign data labeler
+        data_labeler_options = self.options.structured_options.data_labeler
+        if data_labeler_options.is_enabled \
+                and data_labeler_options.data_labeler_object is None:
+
+            try:
+
+                data_labeler = DataLabeler(
+                    labeler_type='structured',
+                    dirpath=data_labeler_options.data_labeler_dirpath,
+                    load_options=None)
+                self.options.set(
+                    {'data_labeler.data_labeler_object': data_labeler})
+
+            except Exception as e:
+                utils.warn_on_profile('data_labeler', e)
+                self.options.set({'data_labeler.is_enabled': False})
```

Comment:
I don't think this belongs in here, we don't instantiate anything in this function for good reason. It's `preprocessing` -> `processing` -> `postprocessing`, BUT the `processing` component isn't managed in here. IMO this convolutes the already hard to follow function

Reply:
Technically this allows a profile to be instantiated without creating the labeler until profile time making it faster. Not sure if this is desirable.

However, where it was before puts the `processing` just before preprocessing anyways. In an ideal state, the data_labeler wouldn't be here anywhere in profiler anyways.